trlx/sweep.py

```diff
@@ -1,83 +1,317 @@
+                    title=f"{target_metric} v. Index",
+                    x="Index",
+                    y=target_metric,
+                    running_ymin=True,
+                    font_size="small",
+                    layout={"x": 6, "y": 5, "w": 6 * 2, "h": 4 * 2},
+                ),
+            ],
+            runsets=[
+                wb.Runset(project=project_name).set_filters_with_python_expr(f'group == "{group_name}"'),
+            ],
+        ),
+    ]
+
+    entity_project = f"{entity_name}/{project_name}" if entity_name else project_name
+    api = wandb.Api()
+    runs = api.runs(entity_project)
+
+    for run in runs:
+        if run.group == group_name:
+            history = run.history()
+            metrics = history.columns
+            break
+
+    metrics = [metric for metric in metrics if not metric.startswith("_")]
+
+    line_plot_panels = []
+    for metric in metrics:
+        line_plot_panels.append(
+            wb.LinePlot(
+                title=f"{metric}",
+                x="Step",
+                y=[f"{metric}"],
+                title_x="Step",
+                smoothing_show_original=True,
+                max_runs_to_show=100,
+                plot_type="line",
+                font_size="auto",
+                legend_position="north",
+            )
+        )
+
+    report.blocks = report.blocks + [
+        wb.H1(text="Metrics"),
+        wb.PanelGrid(
+            panels=line_plot_panels,
+            runsets=[wb.Runset(project=project_name).set_filters_with_python_expr(f'group == "{group_name}"')],
+        ),
+    ]
+
+    if best_config:
+        best_config = best_config["train_loop_config"]
+        config = best_config.pop("default_config")
+        for name, value in best_config.items():
+            *layers, var = name.split(".")
+            if layers:
+                d = config[layers[0]]
+                for layer in layers[1:]:
+                    d = d[layer]
+                d[var] = value
+
+        report.blocks = report.blocks + [
+            wb.H1(text="Best Config"),
+            wb.CodeBlock(code=[json.dumps(config, indent=4)], language="json"),
+        ]
+
+    report.save()
+    print(report.url)
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument("script", type=str, help="Path to the script")
+    parser.add_argument("script", type=str, help="Path to the example script")
     parser.add_argument(
         "--config",
         type=str,
         required=True,
         help="The config file defining the param_space.",
     )
-    parser.add_argument("--num-cpus", type=int, default=4, help="Number of CPUs to use per exp.")
-    parser.add_argument("--num-gpus", type=int, default=1, help="Number of GPUs to use per exp.")
-    parser.add_argument("-y", "--assume-yes", action="store_true", help="Don't ask for confirmation")
     parser.add_argument(
-        "--server-address",
+        "--default_config",
```

Comment:
Do we plan on removing this `default_config`  arg option now that the PR for removing config YAMLs has been merged (#223)? If this is a blocker, I wouldn't be completely opposed to maintaining the leftover configs for now since things are backward-compatible. In a later release, we can fully deprecate config file usage.

Reply:
it's a dilemma whether to convert the rest of configs, or support two ways of updating the "default" one within sweeps. Although I'm already midway with the latter (on the examples which hadn't been converted), it's actually probably more sane to do the former instead.

Reply:
If you're already ahead on the conversions then that's fine too. Let me know if I can help out with them. Which ones still need updating?

Reply:
that would be summarization's, but I think it would make sense to update configs from the updated branch and also after the most recent fix